src/main/java/nl/topicus/mssql2monetdb/CopyTool.java

```diff
@@ -394,196 +484,119 @@ private boolean writeScheduleInfo (Object newValue, int colType)
 		}
 		
 	}
-
+	
 	/**
-	 * Backup a table by copying the data into a backup table named backup_currentTable by
-	 * currentTable and metaData of the MSSQL table that is copied.
-	 * 
-	 * @throws SQLException
+	 * Drops older versions of tables that are no longer used by the view.
+	 * @throws SQLException 
 	 */
-	private void backupTableAndSwitchView(MonetDBTable currentTable) throws SQLException
+	private void dropOldTables(CopyTable table) throws SQLException
 	{
-		LOG.info("Backupping " + currentTable.getToTableSql());
+		LOG.info("Dropping older versions of table '" + table.getToName() + "'...");
 		
-		// create a MonetDBTable object for the backup table
-		MonetDBTable backupTable = new MonetDBTable(currentTable.getCopyTable());
+		Statement q =
+			CopyToolConnectionManager.getInstance().getMonetDbConnection().createStatement();
 		
-		// use the copyTable.toName
-		backupTable.setName(currentTable.getCopyTable().getToName());
-		backupTable.setBackupTable(true);
+		ResultSet result =
+			q.executeQuery("SELECT name FROM sys.tables WHERE name LIKE '" + table.getToName()
+				+ "%' AND name <> '" + table.getToName() + "' "
+				+ "AND schema_id = (SELECT id from sys.schemas WHERE name = '" + table.getSchema()
+				+ "') AND query IS NULL ORDER BY name DESC");
 		
-		// drop and recreate backup table regardless if it exists
-		LOG.info("Drop backup table '" + backupTable.getToTableSql() + "' if exists");
-		MonetDBUtil.dropMonetDBTable(backupTable);
+		int i = 0;
+		int dropCount = 0;
+		Statement stmtDrop =
+			CopyToolConnectionManager.getInstance().getMonetDbConnection().createStatement();
+		while(result.next())
+		{
+			i++;
+			
+			// if table is a fast view-switching table then
+			// 		skip first result -> is current table and referenced by view
+			// 		skip second result -> as backup (TODO: perhaps make this configurable?)
+			if (table.isUseFastViewSwitching())
+				if (i == 1 || i == 2)
+					continue;
+			
+			// build DROP query
+			StringBuilder query = new StringBuilder("DROP TABLE ");
+			
+			if (!StringUtils.isEmpty(table.getSchema()))
+				query.append(MonetDBUtil.quoteMonetDbIdentifier(table.getSchema())).append(".");
+			
+			query.append(MonetDBUtil.quoteMonetDbIdentifier(result.getString("name"))).append(";");
+			
+			// execute DROP query
+			stmtDrop.executeUpdate(query.toString());			
+			dropCount++;
+		}
 		
-		// copy from currentTable to backup table
-		MonetDBUtil.copyMonetDBTableToNewMonetDBTable(currentTable, backupTable);
+		if (i == 0)
+			LOG.info("Table '" + table.getToName() + "' has no older versions");
+		else
```

Comment:
if fast view-switching, then the first and second result is also skipped, resulting in dropCount remaining 0. So it would be better if you also consider these in the Log info so that there will be no info such as Dropped 0 older version. Could be confusing